src/ucp/core/ucp_mm.c

```diff
@@ -111,9 +114,27 @@ ucs_status_t ucp_mem_rereg_mds(ucp_context_h context, ucp_md_map_t reg_md_map,
             } else {
                 ucs_assert(address && length);
 
+                requested_reg_length  = length;
+                requested_reg_address = address;
+
+                if (md_attr->cap.detect_mem_types & UCS_BIT(mem_type)) {
+                    mem_attr.field_mask   = UCT_MD_MEM_ATTR_FIELD_BASE_ADDRESS |
+                                            UCT_MD_MEM_ATTR_FIELD_ALLOC_LENGTH;
```

Comment:
but ib md can't detect Cuda memory

Reply:
Does it make sense to see if md can register given memory, then search for the md that can detect the given memory and get back attributes and use that to register? Based on this https://github.com/openucx/ucx/pull/6178#discussion_r562901684 if we get this back during memory detection then we have to pass those through all the UCP layers.